transports/relay/src/transport.rs

```diff
@@ -106,7 +110,8 @@ where
     }
 
     // Relay to destination over any available relay node.
-    fn relay_to(self, destination: &Peer) -> Result<impl Future<Item=(T::Output, T::MultiaddrFuture), Error=io::Error>, Self> {
+    fn relay_to(&self, destination: &Peer)
+        -> Result<impl Future<Item=(T::Output, T::MultiaddrFuture), Error=io::Error>, io::Error> {
```

Comment:
This format looks weird. While long, I would rather leave it on the same line. Once we can add type aliases for existentials, we can shorten it.